src/naturallanguage.cs

```diff
@@ -0,0 +1,246 @@
+		NSString[] _TagSchemes { get; }
+
+		[Wrap ("Array.ConvertAll (_TagSchemes, e => NLTagSchemeExtensions.GetValue (e))")]
+		NLTagScheme[] TagSchemes { get; }
+
+		[NullAllowed, Export ("string", ArgumentSemantic.Retain)]
+		string String { get; set; }
+
+		[Static]
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
+		[Export ("availableTagSchemesForUnit:language:")]
+		NSString[] GetAvailableTagSchemes (NLTokenUnit unit, NSString language);
+
+		[Static]
+		[Wrap ("Array.ConvertAll (GetAvailableTagSchemes (unit, language.GetConstant()), e => NLTagSchemeExtensions.GetValue (e))")]
+		NLTagScheme[] GetAvailableTagSchemes (NLTokenUnit unit, NLLanguage language);
+
+		[Export ("tokenRangeAtIndex:unit:")]
+		NSRange GetTokenRange (nuint characterIndex, NSString unit);
+
+		[Internal]
+		[NullAllowed, Export ("dominantLanguage")]
+		NSString _DominantLanguage { get; }
+
+		[Wrap ("NLLanguageExtensions.GetValue (_DominantLanguage)")]
+		NLLanguage DominantLanguage { get; }
+
+		[Async (ResultTypeName="NLTaggerEnumerateTagsContinuation")]
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
+		[Export ("enumerateTagsInRange:unit:scheme:options:usingBlock:")]
+		void EnumerateTags (NSRange range, NLTokenUnit unit, NSString scheme, NLTaggerOptions options, Action<NSString, NSRange, bool> block);
+
+		[Wrap ("EnumerateTags (range, unit, scheme.GetConstant (), options, block)")]
+		void EnumerateTags (NSRange range, NLTokenUnit unit, NLTagScheme scheme, NLTaggerOptions options, Action<NSString, NSRange, bool> block);
+
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
+		[Export ("tagAtIndex:unit:scheme:tokenRange:")]
+		[return: NullAllowed]
+		NSString GetTag (nuint characterIndex, NLTokenUnit unit, NSString scheme, [NullAllowed] NSRange tokenRange);
+
+		[Wrap ("NLTagExtensions.GetValue (GetTag (characterIndex, unit, scheme.GetConstant (), tokenRange))")]
+		NLTag GetTag (nuint characterIndex, NLTokenUnit unit, NLTagScheme scheme, [NullAllowed] NSRange tokenRange);
+
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
+		[Export ("tagsInRange:unit:scheme:options:tokenRanges:")]
+		NSString[] GetTags (NSRange range, NLTokenUnit unit, NSString scheme, NLTaggerOptions options, [NullAllowed] out NSValue[] tokenRanges);
+
+		[Wrap ("Array.ConvertAll (GetTags (range, unit, scheme.GetConstant (), options, out tokenRanges), e => NLTagExtensions.GetValue (e))")]
+		NLTag[] GetTags (NSRange range, NLTokenUnit unit, NLTagScheme scheme, NLTaggerOptions options, [NullAllowed] out NSValue[] tokenRanges);
+
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
+		[Export ("setLanguage:range:")]
+		void SetLanguage (NSString language, NSRange range);
+
+		[Wrap ("SetLanguage (language.GetConstant (), range)")]
+		void SetLanguage (NLLanguage language, NSRange range);
+
+		[Export ("setOrthography:range:")]
+		void SetOrthography (NSOrthography orthography, NSRange range);
+
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
```

Comment:
`[EditorBrowsable (EditorBrowsableState.Advanced)]` is default in VS4Mac so maybe `[Protected]` would work better out of the box

Reply:
It is on VS4Mac, but on windows? Protected would mean that the if the user needs the method it will need to subclass. I want to expose the method in case we ever miss an enum value, so that the user can pass an NSString. I would personally leave it like it is there.